### PR TITLE
Add a complementary package for `zod` contracts

### LINKED ISSUE #117: Integration for `zod`
Provide a complementary package for `zod` to use as contract applier.

### PR DESCRIPTION
**Pull request by Drevoed**
resolves #117

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -79,6 +79,7 @@
     "react-dom": "18.2.0",
     "regenerator-runtime": "0.13.7",
     "runtypes": "^6.5.1",
-    "solid-js": "1.4.8"
+    "solid-js": "1.4.8",
+    "tslib": "^2.3.0"
```
Comment (igorkamyshev) [inline package.json:83]: Why do we need `tslib` for this integration? 🤔

File: .changeset/fresh-years-float.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@farfetched/zod': patch
+---
+
+add zod contract package
```
Comment (igorkamyshev) [inline .changeset/fresh-years-float.md:5]:
```suggestion
Add a complementary package for `zod` contracts
```

File: .changeset/fresh-years-float.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@farfetched/zod': patch
```
Comment (igorkamyshev) [inline .changeset/fresh-years-float.md:2]: patch will make release version 0.0.2, I assume it's better to release it as 0.1.0

```suggestion
'@farfetched/zod': minor
```

File: packages/zod/src/__tests__/contract.test.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { string } from 'zod';
+
+import { zodContract } from '../zod_contract';
+
+describe('runtypes/runtypeContract short', () => {
```
Comment (igorkamyshev) [inline packages/zod/src/__tests__/contract.test.ts:5]: Could you add some more tests:
- for `.isData` method, now only `.getErrorMessages` is covered
- for more complex cases, like arrays or records

File: packages/zod/src/__tests__/contract.type_spec.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { expectType } from 'tsd';
+import { Contract } from '@farfetched/core';
+import { string } from 'zod';
+
+import { zodContract } from '../zod_contract';
+
+const contract = zodContract(string());
+
+expectType<Contract<unknown, string>>(contract);
```
Comment (igorkamyshev) [inline packages/zod/src/__tests__/contract.type_spec.ts:9]: Same here, I would like to have tests for type inference in more complex cases like records, arrays, optional fields.

**Review by igorkamyshev — CHANGES_REQUESTED** [review]
Thanks a lot 💙

However, I think, we have to provide at least minimal documentation before release. Could you add it?

File: packages/zod/.babelrc [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+  "presets": [
+    [
+      "@nrwl/web/babel",
+      {
+        "useBuiltIns": "usage"
```
Comment (igorkamyshev) [inline packages/zod/.babelrc:6]: In other packages, we use `false` in the same case. Shall we change it there?

File: packages/zod/jest.config.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable */
+export default {
```
Comment (igorkamyshev) [inline packages/zod/jest.config.ts:2]: 😢 sorry for that, during this PR development I have replaces jest by vitetest. Could you migrate your package as well?

File: packages/zod/project.json [hunk 8]
```diff
@@ -0,0 +1,50 @@
+{
+  "$schema": "../../node_modules/nx/schemas/project-schema.json",
+  "sourceRoot": "packages/zod/src",
+  "projectType": "library",
+  "targets": {
+    "build": {
+      "executor": "@nrwl/js:tsc",
```
Comment (igorkamyshev) [inline packages/zod/project.json:7]: Also, you have migrate all builds from tsc to rollup. Could we do the same for this package?

File: packages/zod/project.json [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+      "outputs": ["{options.outputPath}"],
+      "options": {
+        "outputPath": "dist/packages/zod",
+        "main": "packages/zod/src/index.ts",
+        "tsConfig": "packages/zod/tsconfig.lib.json",
```
Comment (igorkamyshev) [inline packages/zod/project.json:12]: Due to removing jest and migrating to rollup, it became unnecessary to have two different tsconfig filed. We can merge it to single one.

File: packages/zod/project.json [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+      "options": {
+        "lintFilePatterns": ["packages/zod/**/*.ts"]
+      }
+    },
+    "typetest": {
+      "executor": "./tools/executors/tsd:tsd",
+      "inputs": ["{projectRoot}/**/*.type_spec.ts", "{projectRoot}/**/*.ts"]
+    },
+    "test": {
+      "executor": "@nrwl/jest:jest",
```
Comment (igorkamyshev) [inline packages/zod/project.json:28]: Here we have to replace executor with vitest as well.

**Review by igorkamyshev — CHANGES_REQUESTED** [review]
Thanks for the tests 🥹

I'm sorry that during PR development there are plenty of changes in repo tooling, tsc was replaced by rollup and jest was replaced by vitest. We have to support this changes in this PR before merging.

**Comment by Drevoed** [comment]
Sure, will rebase, should not be too bad

**Review by igorkamyshev — COMMENTED** [review]
Great job 💙

I guess, we will be ready to release it after minimal documentation provision 🤗

**Review by igorkamyshev — APPROVED** [review]
I've added links for docs to the site. Let's release it!

**Pull request merged**